src/oop/static_modifier/medium/q5/MainTest.java

```diff
@@ -0,0 +1,114 @@
+package oop.static_modifier.medium.q5;
+
+import global.ObjectTest;
+import global.variables.Clause;
+import global.variables.clauses.NewLine;
+import global.variables.clauses.StringLiteral;
+import global.variables.wrappers.Optional;
+import org.junit.jupiter.api.*;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+
+import static global.tools.CustomAssertions._assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+
+public class MainTest {
+    ObjectTest testClass;
+    ObjectTest outputClass;
+    String className = "Hat";
+    String staticAttributeName = "companyName";
+    String attributeName = "colour";
+    String initialName = "Nike";
+    String initialColour = "Green";
+    String testClassName = "Test";
+
+    @BeforeEach
+    public void setup() throws Throwable {
+        String classString = "oop.static_modifier.medium.q5." + className;
+        testClass = new ObjectTest(classString);
+        String outputClassString = "oop.static_modifier.medium.q5." + testClassName;
+        outputClass = new ObjectTest(outputClassString);
+        String modificationErrorMessage = "You have modified the provided portions of " + className + " class for %s attribute. Please revert them to the original state.";
+        assertTrue(testClass.hasField(attributeName, String.class),
+                String.format(modificationErrorMessage, attributeName));
+        assertTrue(testClass.hasModifier(attributeName, "private"),
+                String.format(modificationErrorMessage, attributeName));
+        String[] test = new String[]{"Black", "Blue", "Red", "Brown"};
+        for (int i = 0; i < test.length; i++) {
+            Object[][] arguments = {
+                    {test[i], String.class}
+            };
+            Object classInstance = testClass.createInstance(arguments);
+            _assertEquals(test[i], testClass.getFieldValue(classInstance, attributeName),
+                    String.format(modificationErrorMessage, attributeName));
+        }
+    }
+
+    @Test
+    @Tag("dependency1")
+    @Order(1)
+    public void correctTestClass() throws Throwable {
+        Object classInstance = outputClass.createInstance();
+        String[] methodModifiers = {"public"};
+        Object[][] arguments = {
+                {new String[0], String[].class}
+        };
+        outputClass.callMethod("main", arguments, methodModifiers, classInstance, new Clause[]{
+                new StringLiteral("This is a " + initialName + " that is " + initialColour + " in colour"),
+                new Optional(new StringLiteral(" ")),
+                new Optional(new NewLine())
+        });
+    }
+
+    @Test
+    @Order(2)
+    public void hatClassHasCorrectAttributes() {
```

Comment:
I think this test should likely come first. You should check here that the value of `companyName` is "Nike" before doing anything else. The test class could change this value for some reason.